BookOfHabits/BookOfHabitsMicroservice.Application.Service/CardsApplicationService.cs

```diff
@@ -0,0 +1,104 @@
+                                                    titleReportField: "Report",
+                                                    tags: "Achievement;Important;Regular;Ordinary",
+                                                    titlePositive: "Healthy",
+                                                    titleNegative: "Damage");
+            var card = new Card(name: new CardName(cardInfo.Name),
+                                options: Domain.Entity.Enums.CardOptions.Status | Domain.Entity.Enums.CardOptions.Value,
+                                titles: tempalteValues,
+                                description: cardInfo.Description);
+            if (cardInfo.Image is not null)
+                card.SetImage(cardInfo.Image);
+            if (cardInfo.TitleCheckElements is not null)
+                card.SetTitlesCheck(cardInfo.TitleCheckElements);
+            card = await cardRepository.AddAsync(card, token)
+                ?? throw new BadRequestException(FormatBadRequestErrorMessage(card.Id, nameof(Card)));
+            return mapper.Map<CardModel>(card);
+        }
+
+        public async Task DeleteCard(Guid id, CancellationToken token = default)
+        {
+            var card = await cardRepository.GetByIdAsync(x => x.Id.Equals(id))
+                ?? throw new NotFoundException(FormatFullNotFoundErrorMessage(id, nameof(Card)));
+            if (await cardRepository.DeleteAsync(card) is false)
+                throw new BadRequestException(FormatBadRequestErrorMessage(id, nameof(Card)));
+        }
+
+        public async Task<IEnumerable<CardModel>> GetAllCardsAsync(CancellationToken token = default)
+        {
+            var cards = (await cardRepository.GetAllAsync(filter: x => x.IsPublic,
+                                                          asNoTracking: true, 
+                                                          cancellationToken: token));
+            return cards.Select(mapper.Map<CardModel>);
+        }
+
+        public async Task<CardModel?> GetCardByIdAsync(Guid id, CancellationToken token = default)
+        {
+            Card card = await cardRepository.GetByIdAsync(x => x.Id.Equals(id),
+                                                                includes: $"{nameof(Card.TemplateValues)}",
+                                                                asNoTracking: true,
+                                                                cancellationToken: token)
+                ?? throw new NotFoundException(FormatFullNotFoundErrorMessage(id, nameof(Card)));
+            return mapper.Map<CardModel>(card);
+        }
+
+        public async Task UpdateCard(UpdateCardModel cardInfo, CancellationToken token = default)
+        {
+            var card = await cardRepository.GetByIdAsync(x => x.Id.Equals(cardInfo.Id), cancellationToken: token)
+                 ?? throw new NotFoundException(FormatFullNotFoundErrorMessage(cardInfo.Id, nameof(Card)));
+            if (cardInfo.Name is not null)
```

Comment:
I have no clue about logic, but the multitude of 'if' it seems like 'smell code' and should be refactored

Reply:
There is no logic here, it is just allowed that there is no data in request entity.
I couldn't find a quick, more concise solution to protect against data erasure if null comes from the front.
I accept recommendations.